Issue #5215: smdba updates

Loquacity opened the pull request:
https://github.com/SUSE/spacewalk/issues/5215

This completes the actual backup section, and technically fulfills the request. Second half of the chapter still needs work, though.

mcalmer commented:
The section about backing up files is a bit misleading. It sounds like "either backup /var/spacewalk OR the list of files and directories below". But that list contains a lot of files which are not in "/var/spacewalk" .
So it is required to back them up also.

We should give this list also to the developers (main on galaxy-devel?) for review. I am not sure if new fetures like "Configuration Channel for salt" or changes in salt ssh and action chains require to backup new paths.

mcalmer commented:
@Loquacity AFAIK you wanted to use just one command for the backup. You forgot one place:

```
. After the initial hot backup has occurred, you can run subsequent hot backups without requiring a database restart:
 +
 
 ----
 smdba backup-hot --backup-dir=/var/spacewalk/db-backup
 ----
```

Here we can use the same command as above and we use with cron. (with --enable=on)

Loquacity commented:
> The section about backing up files is a bit misleading. It sounds like "either backup /var/spacewalk OR the list of files and directories below". But that list contains a lot of files which are not in "/var/spacewalk" .
> So it is required to back them up also.

That came from the original text (I changed the wording but not the meaning). Should we present this as a single option instead of either/or? So it's "if you want to back up manually, back up /var/spacewalk and all these files ... "?


> We should give this list also to the developers (main on galaxy-devel?) for review. I am not sure if new fetures like "Configuration Channel for salt" or changes in salt ssh and action chains require to backup new paths.

I'll send a note to Galaxy-devel

Loquacity commented on adoc/bp_chap_suma_backup.adoc:
```diff
@@ -163,58 +166,41 @@ Stopping the SUSE Manager database...
-Additionally, we recommend that you consider backing up your database to an NFS share location, which can itself be backed up.
+[NOTE]
+====
+When [command]``smdba`` is run for the first time, or if you have changed the location of the database, it will need to reboot your database after performing the archive.
```
[NOTE]
====
When [command]``smdba`` is run for the first time, or if you have changed the location of the backup, it will need to restart your database before performing the archive.
This will result in a small amount of downtime.
Note that regular database backups will not require any downtime.
====

mcalmer commented:
> That came from the original text (I changed the wording but not the meaning). Should we present this as a single option instead of either/or? So it's "if you want to back up manually, back up /var/spacewalk and all these files ... "?

Yes, the original text seems to be already wrong.

I would say we should name this as single option. But we could mention, that `/var/spacewalk/packages/` could be restored also via a re-sync of all channels if the source repository still exists and if it still contain all packages. But a re-sync might take longer.
I think there are too many "if" in this sentence. So I would not really recommend it.

Loquacity commented:
@jcayouette! Please stop merging before it's done! What's the best way to mark something as not ready yet?

Loquacity commented:
@mcalmer OK, updated to make it just one option, and made the restoration into a procedure. Also added the re-sync note as the last (optional) step to the recovery procedure.

keichwa commented:
Lana Brindley <notifications@github.com> writes:

> @jcayouette! Please stop merging before it's done! What's the best way
> to mark something as not ready yet?

Don't create a PR as long as it is not finished ;)  Just work in the
feature or bug fix branch.

-- 
Karl Eichwalder            SUSE Linux GmbH
PES / Documentation        Maxfeldstraße 5
                           90409 Nürnberg, Germany
GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg)

Loquacity commented:
@keichwa Sure, but I still need people to be able to review it. I don't want to be working in isolation.

keichwa commented:
Lana Brindley <notifications@github.com> writes:

> @keichwa Sure, but I still need people to be able to review it. I don't
> want to be working in isolation.

Ok, I see.  I think we now found a solution (labels).

-- 
Karl Eichwalder            SUSE Linux GmbH
PES / Documentation        Maxfeldstraße 5
                           90409 Nürnberg, Germany
GF: Felix Imendörffer, Jane Smithard, Graham Norton, HRB 21284 (AG Nürnberg)
